syntax = "proto3";

package piscan_pb;

message GeneralRequest {
	int32 handle = 1;

	enum RequestType {
		SCANNER_CONTEXT = 0;
		DEMOD_CONTEXT = 1;
		SYSTEM_INFO = 2;
	}

	RequestType type = 2;
}

message ScannerStateRequest {
	int32 handle = 1;

	enum NewState {
		SCAN = 0;
		HOLD = 1;
		MANUAL = 2;
	}

	NewState state = 2;
	int64 manFreq = 3;
	string manModulation = 4;
	string holdIndex = 5;
}

message DemodRequest {
	int32 handle = 1;

	enum DemodFunc {
		SET_SQUELCH = 0;
		SET_GAIN = 1;
	}

	DemodFunc type = 2;
	float level = 3;
}

message LockoutRequest {
	enum Lockout {
		NONE = 0;		// unlocked
		TIMED = 1;	// timed expiration lockout
		SESSION = 2;	// locked until application stops
		PERSISTENT = 3;	// locked until manually unlocked
	}
	string entryIndex = 1;
	Lockout lockoutType = 2;
	int32 lockoutDuration = 3;
}
